plugins/modules/dcnm_interface.py

```diff
@@ -55,6 +55,15 @@
       - deleted
       - query
     default: merged
+  deploy:
```

Comment:
I am a bit confused here.  Why are we adding a global level `deploy` flag that seems to only control the case where we are marking things for delete?  I thought that we were going to reuse the `deploy` flag under the `config` section to control both create and delete cases.

For example using loopback:

* Flag `deploy` under `config` section controls if loopback config to add/remove loopback is sent to device

Reply:
We have introduced this global flag specifically to deal with deleted and overridden states. In the case of delete, there is an option to specify deleted state without including any config block. In this case there is no way to identify if user really wants to deploy after performing delete. Similarly in the case of overridden state, the config block includes a deploy flag which is applicable to the new config included. But the user has no way to specify if the deleted/defaulted information should be deployed or not. This flag handles these special cases.